### ISSUE TITLE
Color shift on certain videos (Transfer characteristics: sRGB/sYCC)

### ISSUE THREAD

**Issue by kparal** [jliljebl/flowblade#673]
Please include the following information.
1) Flowblade version (Help->About):
2.0.0
2) MLT version (Help->Runtime Environment):
6.13.0
3) Your distribution (Ubuntu, Debian, Mint etc.):
Fedora 30, installed from Flathub
4) If possible, try to provide step-by-step instructions on how to reproduce the issue. From developer point of view this is the easieast way to understand what is happening.:

When I render some certain videos, there's a slight color shift from the original. Those color-shifted videos have something in common, in `mediainfo` output they have:
```
Color range                              : Limited
Transfer characteristics                 : sRGB/sYCC
```
While other videos, which get rendered correctly, have:
```
Color range                              : Limited
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
```
When I render those videos affected by color shift in other tools (I have tested `avidemux`, `ffmpeg`, `kdenlive`, `olive`, `openshot`, `pitivi` and `shotcut`), they are rendered with correct colors, and they either contain `BT.709` in media info or the line is completely missing, but they never contain `sRGB/sYCC` - that's only present with `flowblade` with these certain videos.

I have tested playback with `vlc`, `totem` and `smplayer` on Fedora 30 and `vlc`, `windows media player` and `movies&tv` in Windows 10. The color shift is always there, compared to the original.

All the videos (both source and exported) share the following:
```
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
```

Reproducer:
1.Open FlowBlade
2. Drag&drop source video file into Media
3. Confirm changing current profile to file profile
4. Drag&drop media video and audio into the timeline
5. Click Render tab, select `H.264/.mp4`, keep Use Project Profile enabled, keep all defaults, click Render
6. Compare a suitable frame of the exported video with the source video file, notice the color difference

You easily easily test with this source video:
https://youtu.be/0VLc37lNE_4

I have downloaded the video using:
```
$ youtube-dl 'https://youtu.be/0VLc37lNE_4'
```
or here it is attached:
[ytcolors.zip](https://github.com/jliljebl/flowblade/files/3226435/ytcolors.zip)

The downloaded video has these properties:
```
$ mediainfo ytcolors.mkv 
General
Unique ID                                : 49090912745568244372310243209044651858 (0x24EE90DDDD925FBE81B038C9DAE56F52)
Complete name                            : ytcolors.mkv
Format                                   : Matroska
Format version                           : Version 4
File size                                : 12.0 MiB
Duration                                 : 2 min 9 s
Overall bit rate                         : 779 kb/s
Writing application                      : Lavf58.20.100
Writing library                          : Lavf58.20.100
ErrorDetectionType                       : Per level 1

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : High@L4
Format settings                          : CABAC / 2 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 2 frames
Codec ID                                 : V_MPEG4/ISO/AVC
Duration                                 : 2 min 9 s
Width                                    : 1 920 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 30.000 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Title                                    : ISO Media file produced by Google Inc. Created on: 11/22/2018.
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709

Audio
ID                                       : 2
Format                                   : Opus
Codec ID                                 : A_OPUS
Duration                                 : 2 min 9 s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 48.0 kHz
Bit depth                                : 32 bits
Compression mode                         : Lossy
Language                                 : English
Default                                  : Yes
Forced                                   : No
```

When I re-render it using Flowblade, the renderer video has these properties:
```
$ mediainfo ytcolors-flowblade.mp4 
General
Complete name                            : ytcolors-flowblade.mp4
Format                                   : MPEG-4
Format profile                           : Base Media
Codec ID                                 : isom (isom/iso2/avc1/mp41)
File size                                : 22.3 MiB
Duration                                 : 2 min 9 s
Overall bit rate                         : 1 445 kb/s
Writing application                      : Lavf58.12.100

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : High@L4
Format settings                          : CABAC / 4 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 4 frames
Codec ID                                 : avc1
Codec ID/Info                            : Advanced Video Coding
Duration                                 : 2 min 9 s
Bit rate                                 : 1 118 kb/s
Nominal bit rate                         : 3 000 kb/s
Width                                    : 1 920 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 30.000 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.018
Stream size                              : 17.2 MiB (77%)
Writing library                          : x264 core 155
Encoding settings                        : cabac=1 / ref=3 / deblock=1:0:0 / analyse=0x3:0x113 / me=hex / subme=7 / psy=1 / psy_rd=1.00:0.00 / mixed_ref=1 / me_range=16 / chroma_me=0 / trellis=1 / 8x8dct=1 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=-2 / threads=6 / lookahead_threads=1 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / constrained_intra=0 / bframes=3 / b_pyramid=2 / b_adapt=1 / b_bias=0 / direct=1 / weightb=1 / open_gop=0 / weightp=2 / keyint=250 / keyint_min=25 / scenecut=40 / intra_refresh=0 / rc_lookahead=40 / rc=abr / mbtree=1 / bitrate=3000 / ratetol=1.0 / qcomp=0.60 / qpmin=10 / qpmax=51 / qpstep=4 / ip_ratio=1.40 / aq=1:1.00
Color range                              : Limited
Transfer characteristics                 : sRGB/sYCC
Codec configuration box                  : avcC

Audio
ID                                       : 2
Format                                   : MPEG Audio
Format version                           : Version 1
Format profile                           : Layer 3
Format settings                          : Joint stereo
Codec ID                                 : mp4a-6B
Duration                                 : 2 min 9 s
Duration_LastFrame                       : -1 ms
Bit rate mode                            : Constant
Bit rate                                 : 320 kb/s
Maximum bit rate                         : 3 000 kb/s
Channel(s)                               : 2 channels
Sampling rate                            : 48.0 kHz
Compression mode                         : Lossy
Stream size                              : 4.94 MiB (22%)
Writing library                          : LAMEªEÐ
Default                                  : Yes
Alternate group                          : 1
```

And if you compare e.g. frame at `00:00:26:00`, you see a clear color shift (open the pictures into a full size and switch between them forth and back):
Original:
![src](https://user-images.githubusercontent.com/755451/58464236-a4399e00-8135-11e9-998c-aaf9c515c30e.png)
Flowblade rendered:
![flowblade](https://user-images.githubusercontent.com/755451/58464253-aac81580-8135-11e9-8a83-aae3f36fd0ba.png)

**Comment by jliljebl** [comment]
Thank you for this detailed, thoroughly researched and well presented bug report.

I was able to reproduce the issue, and the data you provide establishes that this is in fact a Flowblade issue.

We have very good prospects of fixing this relatively easily, we are almost certainly ignoring some color space details when creating *mlt.Consumers* for rendering in *renderconsumer.py* module. There are some other possibilities that we may need to look at, maybe we are ignoring giving some parameters when creating timeline or clip producers for particular video producers.

I will attempt to get the fix in 2.2 but the fix may get postponed until after summer as the 2.2 release is quite imminent.

Again, thanks for putting in the work to establish all the facts needed to get this fixed.

**Comment by jliljebl** [comment]
This is now fixed, I was running outdated MLT profiles which caused some color errors. Thanks for bringing this to my attention, I'm not running some technical test grid per release so I sometimes miss stuff but now we're a-okay again.
